### PR TITLE
Added week6 homework

### PR DESCRIPTION
**Pull request by semih1239**

### CODE REVIEW

File: final/MealsharingNet/IReservationRepository.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using MealsharingNet.models;
+
+namespace MealsharingNet;
+
+public interface IReservationRepository
+{
+    Task <IEnumerable<Reservation>> ListReservations();
```
Comment (calindurnea) [inline final/MealsharingNet/IReservationRepository.cs]: You don't have to add a space after Task in these cases `Task <IEnumerable<Reservation>>` => `Task<IEnumerable<Reservation>>`

Comment (calindurnea) [reply final/MealsharingNet/IReservationRepository.cs]: Similar below on line 9 and in the controller

File: final/MealsharingNet/IReviewRepository.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+using MealsharingNet.Models;
+
+namespace MealsharingNet;
+
+public interface IReviewRepository
+{
+    Task <IEnumerable<Review>> ListReviews();
+    Task AddReview(Review review);
+    Task<List<Review>> GetMealReviews(int mealId);
```
Comment (calindurnea) [inline final/MealsharingNet/IReviewRepository.cs]: For consistency,  the parameter here, `mealId`, should be named the same in the ReviewRepository and in the ReviewController. (The same for the Reservations) :)

File: final/MealsharingNet/Models/Meal.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+namespace MealsharingNet.models;
```
Comment (calindurnea) [inline final/MealsharingNet/Models/Meal.cs]: The namespace here should be with uppercase, models => Models.

File: final/MealsharingNet/Models/User.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+namespace MealsharingNet;
```
Comment (calindurnea) [inline final/MealsharingNet/Models/User.cs]: The namespace here should also have `Models` as part of the name

Comment (calindurnea) [reply final/MealsharingNet/Models/User.cs]: MealsharingNet.Models

File: final/MealsharingNet/Program.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+
+builder.Services.AddControllers();
+builder.Services.AddSingleton<IMealRepository, MealRepository>();
+builder.Services.AddSingleton<IReservationRepository, ReservationRepository>();
+builder.Services.AddSingleton<IReviewRepository, ReviewRepository>();
+// builder.Services.AddSingleton<IUserRepository, UserRepository>();
```
Comment (calindurnea) [inline final/MealsharingNet/Program.cs]: Any reason why you chose to comment this part? 😊

**Review by calindurnea — COMMENTED** [review]
Very nicely done 💯 
I left a few comments, let me know if you have any questions or if I should explain something 😊

File: final/MealsharingNet/Program.cs (hunk 5 above)
Comment (semih1239) [reply final/MealsharingNet/Program.cs]: I forget it here :blush:

**Pull request merged**